core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -59,13 +60,14 @@ private BootstrapCheck() {
      * checks
      *
      * @param settings              the current node settings
+     * @param dataDirectoryNumber   the integer number of the directories who's locks under the data directory we were able to take
```

Comment:
I don't think this is a correct javadoc, the first node lock id will be 0, which would correspond to being able to take "zero" directories